NMS-14613  Quick Start: Login and create a user (#5170)
OpenNMS/opennms: docs/modules/operation/pages/deep-dive/user-management/security-roles.adoc

[[ga-role-user-management-roles]]
= Assign User Permissions

Create user permission levels by assigning security roles.
These roles regulate access to the web UI and the REST API to exchange monitoring and inventory information.
ifndef::opennms-prime[]

> In a distributed installation, the Minion instance requires the `ROLE_MINION` permission to interact with {page-component-title}.

Reviewer: ```suggestion
In a distributed installation, Minion instances require an account with the `ROLE_MINION` permission to interact with {page-component-title}.
This can be one account for all Minions to share, or one account per Minion.
```

| Default permissions for a new user to interact with the web UI: can escalate and acknowledge alarms and notifications.
|===

> == Assign roles to users

Reviewer 2: ```suggestion
== Assign roles to users

```

. Log in as a user with admin permissions.
. Click the gear icon at the top-right of the screen.
. Choose menu:Configure OpenNMS[Configure Users, Groups and On-Call Roles > Configure Users].
. Click the modify icon next to the user you want to update.
. Select the role from menu:Security Roles[Available Roles].
. Click *Add* to assign the security role to the user.
. Click *Finish* to apply the changes.
. Log out and log back in to apply the new security role settings.

== Create custom security roles

To create a custom security role you need to define the name and specify the security permissions.

. Create a file called `$\{OPENNMS_HOME}/etc/security-roles.properties`.
. Add a property called `roles`, and for its value, a comma-separated list of the custom security roles as in the following example:

[source, properties]
----
roles=operator,stage
----

> To define permissions associated with the custom security role, manually update the application context of the Spring security at `$\{OPENNMS_HOME}/jetty-webapps/opennms/WEB-INF/applicationContext-spring-security.xml`.

Reviewer 2: ```suggestion
To define permissions associated with the custom security role, manually update the application context of the Spring security at `$\{OPENNMS_HOME}/jetty-webapps/opennms/WEB-INF/applicationContext-spring-security.xml`.

```

. Click *Finish* to apply the changes.
. Log out and log in to apply the new security role settings.

> == Create custom security roles

Reviewer: Suggestion for replacing this line to the end of the file:

```
== Create custom security roles

To create a custom security role you need to define the name and specify the security permissions.

* Create a file called `$\{OPENNMS_HOME}/etc/security-roles.properties`.
* Add a property called `roles`, and for its value, a comma-separated list of the custom security roles, for example:

[source, properties]
----
roles=operator,stage
----

To define permissions associated with the custom security role, manually update the application context of the Spring security at `$\{OPENNMS_HOME}/jetty-webapps/opennms/WEB-INF/applicationContext-spring-security.xml`.
```
